packages/cozy-pouch-link/src/PouchManager.spec.js

```diff
@@ -334,4 +454,63 @@ describe('PouchManager', () => {
       }
     })
   })
+
+  describe('warmupQueries', () => {
+    let manager
+    const executeMock = jest.fn()
+    beforeEach(() => {
+      let newManagerOptions = {
+        ...managerOptions,
+        executeQuery: executeMock,
+        doctypesReplicationOptions: {
+          'io.cozy.todos': {
+            strategy: 'fromRemote',
+            warmupQueries: [query1(), query2()]
+          }
+        }
+      }
+      manager = new PouchManager(['io.cozy.todos'], newManagerOptions)
+    })
+
+    it('should executes warmeupQueries on the first replicationLoop only', async () => {
+      jest
+        .spyOn(rep, 'startReplication')
+        .mockImplementation(() => Promise.resolve({}))
+      await manager.replicateOnce()
+      await sleep(10)
+      executeMock.mockResolvedValue({})
+      expect(executeMock).toHaveBeenCalledTimes(2)
+      expect(executeMock).toHaveBeenNthCalledWith(
+        1,
+        query1()
+          .definition()
+          .toDefinition()
+      )
+      expect(executeMock).toHaveBeenNthCalledWith(
+        2,
+        query2()
+          .definition()
+          .toDefinition()
+      )
+      expect(manager.getPersistedWarmedUpQueries()).toEqual({
+        'io.cozy.todos': ['query1', 'query2']
+      })
+      //Simulation of a loop. Let's replicate again
+      await manager.replicateOnce()
+      await sleep(10)
+      expect(executeMock).toHaveBeenCalledTimes(2)
+    })
+
+    it('should not persist or cache warmedup queries if one have failled', async () => {
+      jest
+        .spyOn(rep, 'startReplication')
+        .mockImplementation(() => Promise.resolve({}))
+      executeMock.mockRejectedValueOnce('error').mockResolvedValue({})
+
+      await manager.replicateOnce()
+      await sleep(10)
```

Comment:
Why do we need to sleep here ? Don't we have access to a promise that we could await instead ?